java/vector/src/main/java/org/apache/arrow/vector/ipc/ArrowFileWriter.java

```diff
@@ -129,12 +128,6 @@ protected void endInternal(WriteChannel out) throws IOException {
   @Override
   protected void ensureDictionariesWritten(DictionaryProvider provider, Set<Long> dictionaryIdsUsed)
       throws IOException {
-    if (dictionariesWritten) {
-      return;
-    }
-    dictionariesWritten = true;
-    // Write out all dictionaries required.
-    // Replacement dictionaries are not supported in the IPC file format.
```

Comment:
Just for my clarification:

Referring to [Apache Arrow Columnar specification](https://arrow.apache.org/docs/format/Columnar.html#ipc-file-format). 
> Further more, it is invalid to have more than one non-delta dictionary batch per dictionary ID (i.e. dictionary replacement is not supported). 

Here the relevant part of the code to provide that functionality has been removed. Does this change that functionality? Does it need a documentation update?